lib/components/Snackbar/index.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { ReactNode, useEffect, useState, HTMLAttributes } from "react";
+import { Text } from "../Typography";
+import "./snackbar.scss";
+import clsx from "clsx";
+import { Button } from "../Button";
+import { LabelPairedXmarkSmBoldIcon } from "@deriv/quill-icons";
+
+interface SnackbarProps extends HTMLAttributes<HTMLDivElement> {
+    icon?: ReactNode;
+    message: string;
+    actionText?: string;
+    hasCloseButton?: boolean;
+    onClose: () => void;
+    onActionClick?: () => void;
+    isOpen: boolean;
+}
+
+export const Snackbar = ({
+    icon: Icon,
+    message,
+    actionText,
+    onActionClick,
+    hasCloseButton = true,
+    isOpen = false,
+    onClose,
+    ...rest
+}: SnackbarProps) => {
+    const animationSpeedObj = Object.freeze({
+        fast: 'fast',
+        slow: 'slow'
+    });
+
+    const [animationSpeed, setAnimationSpeed] = useState<keyof typeof animationSpeedObj>(animationSpeedObj.slow);
+    useEffect(() => {
+        if (isOpen) {
+            setAnimationSpeed(animationSpeedObj.slow);
+            const timer = setTimeout(() => {
+                onClose?.();
+            }, 3000);
+
+            return () => {
+                clearTimeout(timer);
+            };
+        }
+    }, [isOpen]);
+
+    const handleClose = () => {
+        onClose?.();
+        setAnimationSpeed(animationSpeedObj.fast);
+    };
+
+    const handleActionClick = () => {
+        onActionClick?.();
+        setAnimationSpeed(animationSpeedObj.fast);
+    };
+    return (
+        <>
+            {isOpen && (
+                <div
+                    className={clsx(
+                        "snackbar",
+                        animationSpeed === animationSpeedObj.fast
+                            ? "fast-animation"
+                            : "slow-animation",
+                    )}
+                    {...rest}
```

Comment:
Because of the position of `{...rest}` it might overwrite `className`. Maybe it's better to insert `{..rest}` in the top?

Reply:
@henry-deriv  you might have forgotten to address this :)

Reply:
i didnt make any changes because it makes no difference in this case.